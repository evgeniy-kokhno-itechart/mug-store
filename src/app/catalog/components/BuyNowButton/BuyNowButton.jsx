import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { PropTypes } from 'prop-types';
import './BuyNowButton.scss';

class BuyNowButton extends Component {
  handleButtonClick = () => {
    const { product, onBuyNow } = this.props;
    onBuyNow(product);
  };

  render() {
    const { customClasses } = this.props;
    return (
      <button type="button" className={`button button--solid buy-now-button focusable ${customClasses}`} onClick={this.handleButtonClick}>
        <FontAwesomeIcon icon="cart-arrow-down" />
      </button>
    );
  }
}

BuyNowButton.propTypes = {
  customClasses: PropTypes.string,
  product: PropTypes.shape({
    id: PropTypes.string,
    imageURL: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    category: PropTypes.shape({ id: PropTypes.string, name: PropTypes.string }),
    basePrice: PropTypes.number,
    currentCurrencyPrice: PropTypes.number,
    discount: PropTypes.number,
    producer: PropTypes.string,
  }).isRequired,
  onBuyNow: PropTypes.func.isRequired,
};

BuyNowButton.defaultProps = {
  customClasses: '',
};

export default BuyNowButton;
