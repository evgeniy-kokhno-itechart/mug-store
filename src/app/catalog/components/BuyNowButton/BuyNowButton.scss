@import "Variables";

.buy-now-button {
    &,
    &:link,
    &:visited {
        display: inline-block;
        position: relative;
    }

    &:hover {
        animation-name: moveLeftRight;
        animation-duration: 0.2s;
        animation-iteration-count: infinite; 
    }
    
    &:active {
        animation: none;
        &::after {
            background-color: $color-gray-dark;
            transform: scale(2);
            opacity: 0.1;
        }
    } 

    &::after {
        content: "";
        transition: all 0.5s;
        display: inline-block;
        width: 100%;
        height: 100%;
        position: absolute;
        top: 0;
        left: 0;
        z-index: -1;
    } 
}

@keyframes moveLeftRight {
    0% {
        transform: translateY(-0.15rem);
    }

    50% {
        transform: translateY(0.15rem);
    }

    100% {
        transform: translateY(-0.15rem);
    }
} 