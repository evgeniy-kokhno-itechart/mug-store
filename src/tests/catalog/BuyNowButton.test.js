/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BuyNowButton from '../../catalog/components/BuyNowButton';

configure({ adapter: new Adapter() });

describe('<BuyNowButton />', () => {
  let BuyNowButtonWrapper;

  beforeEach(() => {
    const onBuyNowFunc = jest.fn();
    BuyNowButtonWrapper = shallow(
      <BuyNowButton
        onBuyNow={onBuyNowFunc}
        product={{
          id: '',
          imageURL: '',
          title: 'Test Product',
          description: '',
          category: { id: '1', name: 'Test Category' },
          basePrice: 1,
          currentCurrencyPrice: 1,
          discount: 0,
          producer: '',
        }}
      />,
    );
  });

  it('check render without customClasses provided', () => {
    expect(BuyNowButtonWrapper.debug()).toMatchSnapshot();
  });

  it('check render with customClasses = testclass provided', () => {
    BuyNowButtonWrapper.setProps({ customClasses: 'testclass' });
    expect(BuyNowButtonWrapper.debug()).toMatchSnapshot();
  });
});
