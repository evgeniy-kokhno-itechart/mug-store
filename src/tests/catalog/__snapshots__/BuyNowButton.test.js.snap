// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuyNowButton /> check render with customClasses = testclass provided 1`] = `
<button
  className="btn btn-secondary testclass"
  onClick={[Function]}
  type="button"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="cart-arrow-down"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</button>
`;

exports[`<BuyNowButton /> check render without customClasses provided 1`] = `
<button
  className="btn btn-secondary "
  onClick={[Function]}
  type="button"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="cart-arrow-down"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</button>
`;
